diag_manager/diag_manager.F90

```diff
@@ -584,14 +590,26 @@ INTEGER FUNCTION register_diag_field_array(module_name, field_name, axes, init_t
     END IF
 
     if (use_modern_diag) then
-            call diag_object_placeholder(1)%register &
-       (module_name, field_name, axes, init_time, &
-       long_name, units, missing_value, Range, mask_variant, standard_name, &
-       do_not_log, err_msg, interp_method, tile_count, area, volume, realm) !(no metadata here)
+      !! Create a diag object, initialize it with the registered data, and insert
+      !! it ino the diag_obj_container singleton.
+
+      allocate( diag_obj )
```

Comment:
Where is this deallocated?

Reply:
Contents of the container are deallocated by the container.
Either by the container destructor or the user calling the clear() function.